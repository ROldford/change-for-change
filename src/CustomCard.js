import React from 'react';
import PropTypes from 'prop-types';
import {Card, CardBody, CardTitle, CardText} from 'reactstrap';
import Image from 'react-graceful-image';

let CustomCard = (props) => {
  const data = props.data;
  return (
    <Card className="m-2">
      <CardBody>
        <CardTitle>
          {data.cardTitle}
          <br />
          <Image
            src={data.imageSource}
            width="64"
            height="64"
            alt={data.imageAltText}
          />
        </CardTitle>
        <CardText>{data.cardText}</CardText>
      </CardBody>
    </Card>
  );
}

CustomCard.propTypes = {
  data: PropTypes.shape({
    cardTitle: PropTypes.string,
    cardText: PropTypes.string,
    imageSource: PropTypes.string,
    imageAltText: PropTypes.string
  })
}

export default CustomCard;

    // {/*getWrapDivs(props.position)*/}
