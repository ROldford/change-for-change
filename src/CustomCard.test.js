import CustomCard from "./CustomCard";

it("renders correctly", () => {
  const data = {
    cardTitle: "Test Card",
    cardText: "Test card text goes here",
    imageSource: "./testdata/placeholder.png",
    imageAltText: "Test alt text"
  }
  const wrapper = shallow(
    <CustomCard data={data} />
  );
  expect(wrapper).toMatchSnapshot();
});
