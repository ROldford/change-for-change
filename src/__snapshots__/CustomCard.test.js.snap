// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Card
  className="m-2"
  tag="div"
>
  <CardBody
    tag="div"
  >
    <CardTitle
      tag="h5"
    >
      Test Card
      <br />
      <GracefulImage
        alt="Test alt text"
        className={null}
        height="64"
        noLazyLoad={false}
        noPlaceholder={false}
        noRetry={false}
        placeholderColor="#eee"
        retry={
          Object {
            "accumulate": "multiply",
            "count": 8,
            "delay": 2,
          }
        }
        src="./testdata/placeholder.png"
        style={Object {}}
        width="64"
      />
    </CardTitle>
    <CardText
      tag="p"
    >
      Test card text goes here
    </CardText>
  </CardBody>
</Card>
`;
